ui/src/treeDataProvider/AzureAccountTreeItem.ts

```diff
@@ -0,0 +1,171 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import { Disposable, Extension, extensions } from 'vscode';
+import * as vscode from 'vscode';
+import * as types from '../../index';
+import { AzureAccount, AzureLoginStatus, AzureResourceFilter } from '../azure-account.api';
+import { localize } from '../localize';
+import { TestAzureAccount } from '../TestAzureAccount';
+import { nonNullProp, nonNullValue } from '../utils/nonNull';
+import { AzureWizardPromptStep } from '../wizard/AzureWizardPromptStep';
+import { AzExtParentTreeItem } from './AzExtParentTreeItem';
+import { AzExtTreeItem } from './AzExtTreeItem';
+import { GenericTreeItem } from './GenericTreeItem';
+import { SubscriptionTreeItem } from './SubscriptionTreeItem';
+
+const signInLabel: string = localize('signInLabel', 'Sign in to Azure...');
+const createAccountLabel: string = localize('createAccountLabel', 'Create a Free Azure Account...');
+const selectSubscriptionsLabel: string = localize('noSubscriptions', 'Select Subscriptions...');
+const signInCommandId: string = 'azure-account.login';
+const createAccountCommandId: string = 'azure-account.createAccount';
+const selectSubscriptionsCommandId: string = 'azure-account.selectSubscriptions';
+
+type SubscriptionTreeItemType = { new(parent: AzExtParentTreeItem, root: types.ISubscriptionRoot): SubscriptionTreeItem };
+
+export class AzureAccountTreeItem extends AzExtParentTreeItem implements types.AzureAccountTreeItem {
+    public static contextValue: string = 'azureextensionui.azureAccount';
+    public readonly contextValue: string = AzureAccountTreeItem.contextValue;
+    public readonly label: string = 'Azure';
+    public readonly childTypeLabel: string = localize('subscription', 'subscription');
+    public autoSelectInTreeItemPicker: boolean = true;
+    public disposables: Disposable[] = [];
+
+    private _subscriptionTreeItemType: SubscriptionTreeItemType;
+    private _azureAccount: AzureAccount;
+    private _subscriptionTreeItems: SubscriptionTreeItem[] | undefined;
+
+    constructor(parent: AzExtParentTreeItem | undefined, subscriptionTreeItemType: SubscriptionTreeItemType, testAccount?: TestAzureAccount) {
```

Comment:
In the readme you refer to passing in a "subscriptionTreeItem" (here the param is named subscriptionTreeItemType)